app/models/katello/glue/candlepin/repository.rb

```diff
@@ -18,7 +18,7 @@ def should_update_content?
       def yum_gpg_key_url
         # if the repo has a gpg key return a url to access it
         if (gpg_key && gpg_key.content.present?)
-          host = Facter.value(:fqdn) || SETTINGS[:fqdn]
+          host = SETTINGS[:fqdn]
           gpg_key_content_api_repository_url(self, :host => host + "/katello", :protocol => 'https')
```

Comment:
Would it make sense to have a dedicated setting for this? In the installer we explicitly set the `post_sync_url` and we have the Foreman URL. That would allow running behind a load balancer easier.

Reply:
If the load balancer setting will be different than the :fqdn setting, i think that makes perfect sense :+1:    Should we have some generic value like :external_fqdn ?

Reply:
big :+1: for generic setting like `external_fqdn` or `external_dns_name` that controls all that stuff. There are instances where you'd want to be able to override stuff like the `post_sync_url`, but IMO that could all be hidden in custom-hiera.

Reply:
Yeah, i can think of at least ~4 potential uses of an external_fqdn within katello

Reply:
Turns we already have `Setting[:foreman_url]` which is probably a good starting point.